payments-core/src/main/java/com/stripe/android/googlepaylauncher/GooglePayLauncher.kt

```diff
@@ -276,7 +281,54 @@ class GooglePayLauncher internal constructor(
         fun onResult(result: Result)
     }
 
-    internal companion object {
+    companion object {
         internal const val PRODUCT_USAGE = "GooglePayLauncher"
+
+        /**
+         * Create a [GooglePayLauncher] used for Jetpack Compose.
+         *
+         * This API uses Compose specific API [rememberLauncherForActivityResult] to register a
+         * [ActivityResultLauncher] into current activity, it should be called as part of Compose
+         * initialization path.
+         * The GooglePayLauncher created is remembered across recompositions. Recomposition will
+         * always return the value produced by composition.
+         */
+        @Composable
+        fun rememberLauncher(
+            config: Config,
+            readyCallback: ReadyCallback,
+            resultCallback: ResultCallback
+        ): GooglePayLauncher {
+            val context = LocalContext.current
+            val lifecycleScope = LocalLifecycleOwner.current.lifecycleScope
+            val activityResultLauncher = rememberLauncherForActivityResult(
+                GooglePayLauncherContract(),
+                resultCallback::onResult
+            )
+
+            return remember {
```

Comment:
Should this be keyed off the `config` and `readyCallback`? So that in case a different instance of either of those parameters are provided on a subsequent composition, a new instance is returned using those new values.

```kotlin
return remember(config, readyCallback) {
    // ...
}
```

I think the same question would apply to `rememberLauncher` also. (I could be wrong – still a bit of a Compose novice!)

Reply:
I'm also pretty new to Compose. You're partially right, I've updated the example to use the `config` as a key. For the callbacks we use `rememberUpdatedState`, which is also what [`rememberLauncherForActivityResult`](https://androidx.tech/artifacts/activity/activity-compose/1.3.0-alpha07-source/androidx/activity/compose/ActivityResultRegistry.kt.html) uses. I added you to the API review doc, ptal and let me know if you have any suggestions.